### ISSUE TITLE
Activerecord associations (:includes) not working

### ISSUE THREAD

**Issue by bgvo** [michaelherold/ksuid-ruby#26]
This happens when trying to eager load associations:

```ruby
Organization.includes(:users).each do |o|
    ...
 end
```

Throws:
```ruby
NoMethodError: undefined method `first' for nil:NilClass
```

The problem is in:
`activerecord-6.1.4.1/lib/active_record/associations/preloader/association.rb:122:in `block in records_for'`

Specifically, this line:

```ruby
owner = owners_by_key[convert_key(record[association_key_name])].first
```

from

```ruby
        def records_for(ids)
          scope.where(association_key_name => ids).load do |record|
            # Processing only the first owner
            # because the record is modified but not an owner

            owner = owners_by_key[convert_key(record[association_key_name])].first
            association = owner.association(reflection.name)
            association.set_inverse_instance(record)
          end
        end
```

In order to lookup associated records, Activerecord looks for the key stored as KSUID::Type. For instance, value of `owners_by_key`:

```ruby
{<KSUID(1ym0xC8YYVoho8JmB19Y5Rsdi3O)>=>[#<Organization id: <KSUID(1ym0xC8YYVoho8JmB19Y5Rsdi3O)>...}
```

And the value for `convert_key(record[association_key_name])`:

```ruby
<KSUID(1ym0xC8YYVoho8JmB19Y5Rsdi3O)>
```

It's the same id. But when trying to fetch that key the result is nil. I was assuming it had to do with how KSUID converts to object_id, as the two 'same' KSUIDs convert to different object_ids. Leading to not finding the key as the two object_ids are different.

I tried to define `#object_id` in KSUID::Type to return a static value:

```ruby
    def object_id
      self.to_i
    end
```

However, that doesn't fix the issue and still, the key is not found in Activerecord Association's class' `owners_by_key` hash.

Any ideas how to fix this?

**Comment by stormmaster42** [comment]
Just investigated this further.
Patching the KSUID::Type with
`    # returns true if obj and other refer to the same hash key.
    # original implementation compares object eqality ("==")
    # https://apidock.com/ruby/Object/eql%3F
    def eql?(other)
      hash == other.hash
    end

    # original implementation uses object id,
    # so even if the value was the same, the object wouldn't be recognized
    # to be the same hash key
    # https://apidock.com/ruby/Object/hash
    delegate :hash, to: :raw
`
fixes the issues with preload/includes

**Comment by michaelherold** [comment]
This should be fixed in the newly released [v0.4.0](https://rubygems.org/gems/ksuid/versions/0.4.0). Please let me know if you're still seeing the broken behavior!
